src/java.desktop/windows/native/libawt/windows/awt_Win32GraphicsEnv.cpp

```diff
@@ -123,9 +123,10 @@ BOOL DWMIsCompositionEnabled() {
     }
 
     dwmIsCompositionEnabled = bRes;
+    jboolean ignoreException;
 
     JNIEnv *env = (JNIEnv *)JNU_GetEnv(jvm, JNI_VERSION_1_2);
-    JNU_CallStaticMethodByName(env, NULL,
+    JNU_CallStaticMethodByName(env, &ignoreException,
```

Comment:
As far as understand from the previous fix the "ignoreException" parameter does not suppress the currently raised exception, it just suppresses the warning in the XCheck:jni. So it will be good to propagate an exception to the user if this code is executed on the "java" thread, and log it by the trace macro if it is executed on a toolkit thread.

Reply:
Yeah this looks like the wrong fix.
-Xcheck:jni only warns like this if there are actual exceptions, doesn't it ?
So if we had one, where was it ?

Note that Windows Server 2019 is a staple of the CI testing at Oracle and the test hasn't failed in the > 2 months since it was integrated, so I'm wondering why it fails in your case ?
Was this a debug build perhaps ?

Reply:
I think Xcheck:jni raises a warning when two JNI calls are made in a row w/o calling exception check in between. So it is not necessary to have an actual exception to produce a warning.

Probably it is reproduced there, because that system is "true" headless, and the execution code path is just different, or something like that, need to check what is the next/prev JNI call.

Reply:
Hello,  in our central tests it was indeed a fastdbg OpenJDK.   On my local Windows 10 machine I could not reproduce it.
Our central tests run with -Djava.awt.headless=true , but I think the Win2019 server machine itself is not headless .

Reply:
I could add a  J2dTraceLn(J2D_TRACE_ERROR, . . . )  in case an exception occurs.  This is already done at some other places in  DWMIsCompositionEnabled so it might fit better than just ignoring the exception.

Reply:
Before ignoring an exception it is better to check why the warning was raised, if that warning is correct then we should try to propogate/report exception to the app, if it is not possible then we should use trace macro.

Reply:
Hello, I added a trace call.  This is similar to what is done in https://github.com/openjdk/jdk/blob/master/src/java.desktop/windows/native/libawt/java2d/d3d/D3DRenderQueue.cpp#L869   .

Reply:
In the old fix the trace was added there because that method is executed in the infinite loop on some special thread, so an exception could not be thrown to the user, is it the case in the current issue?